introducing state and props

lauradurandv opened the pull request.

xjqiu28 commented:
Good job on the homework! Your app is working perfectly with some tweaking needed.

1) You are missing the reset form after you press on submit.  Would you happen to know how to add that functionality in?

2) Just a suggestion, you don't need to do it if you don't want to. For the input element, you can put something called a [placeholder](https://www.w3schools.com/tags/att_input_placeholder.asp), just so instead of having an empty input box, it can say something like "please enter to do here" when you reset the form after clicking on submit.

3) This is also a good to know but not necessary. For your props, there is something called [destructuring](https://handsonreact.com/docs/props). According to the website, it is generally a good practice and makes the code easier to read!

Other than that, everything is working and hope you have a great rest of the week! 😄

lauradurandv commented:
Thanks. I've just added in those changes. I am not sure if I implemented
the reset the best way but would love your feedback! I also really
appreciate the extra info I also added those in.
Here is my pull request link:
https://github.com/lauradurandv/react-todo/pull/3


On Sat, Mar 9, 2024 at 10:54 PM xjqiu28 ***@***.***> wrote:

> Good job on the homework! Your app is working perfectly with some tweaking
> needed.
>
>    1.
>
>    You are missing the reset form after you press on submit. Would you
>    happen to know how to add that functionality in?
>    2.
>
>    Just a suggestion, you don't need to do it if you don't want to. For
>    the input element, you can put something called a placeholder
>    <https://www.w3schools.com/tags/att_input_placeholder.asp>, just so
>    instead of having an empty input box, it can say something like "please
>    enter to do here" when you reset the form after clicking on submit.
>    3.
>
>    This is also a good to know but not necessary. For your props, there
>    is something called destructuring <https://handsonreact.com/docs/props>.
>    According to the website, it is generally a good practice and makes the
>    code easier to read!
>
> Other than that, everything is working and hope you have a great rest of
> the week! 😄
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lauradurandv/react-todo/pull/3#issuecomment-1987060191>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGJKOKACFK4JC7NI7Z7E7EDYXPKRBAVCNFSM6AAAAABEOGM32WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBXGA3DAMJZGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

xjqiu28 commented:
I see all the changes!! 

1) You can remove the title on line 3 of ToDoListItem.jsx, since you don't need it and you only passed down one prop anyway, which is todo, from TodoList.jsx. I will send you an example on why destructuring is cool.

2) I just realized that the title for your html is Vite+React, can you change it to To Do List?

3) There are a lot of ways to do reset and yours is working fine, so don't worry about it.
